src/abe_policy/dimension.rs

```diff
@@ -223,15 +265,14 @@ impl Dimension {
     /// Returns an error if the operation is not permitted or if the attribute
     /// is not found.
     pub fn remove_attribute(&mut self, attr_name: &AttributeName) -> Result<(), Error> {
-        if self.order.is_some() {
-            Err(Error::OperationNotPermitted(
-                "Hierarchical dimension are immutable".to_string(),
-            ))
-        } else {
-            self.attributes
+        match self {
+            Self::Unordered(attributes) => attributes
                 .remove(attr_name)
                 .map(|_| ())
```

Comment:
Useless line. Juste ignore the returned value by not returning this expression's value.

Reply:
So we will not raise an exception if the Attribute wasn't removed?